products/sentinel_one.py

```diff
@@ -153,15 +99,136 @@ def _authenticate(self):
+        site_ids = (site_id) if site_id else list()
+        account_ids = (account_id) if account_id else list()
+        account_names = (account_name) if account_name else list()
+
+        if not site_ids and not account_ids and not account_names:
+            # extract account/site ID from configuration if set
+            if 'account_id' in config[self.profile]:
+                for id in config[self.profile]['account_id'].split(','):
+                    if id not in account_ids:
+                        account_ids.append(id.strip())
+
+            if 'site_id' in config[self.profile]:
+                for id in config[self.profile]['site_id'].split(','):
+                    if id not in site_ids:
+                        site_ids.append(id.strip())
+
+            if 'account_name' in config[self.profile]:
+                for name in config[self.profile]['account_name'].split(','):
+                    if name not in account_names:
+                        account_names.append(name.strip())
+
+        # determine site and account IDs to query (default is all)
+        self._site_ids = list()
+        self._account_ids = list()
+
+        if account_ids: # verify provided account IDs are valid
+            # create batch of 10 account IDs per call
+            counter = 0
+            temp_list = []
+            i = 0
+            while i < len(account_ids):
+                temp_list.append(account_ids[i])
+                counter += 1
+                if counter == 10 or i == len(account_ids) - 1:
+                    response = self._get_all_paginated_data(self._build_url(f'/web/api/v2.1/accounts'),
+                                                            params={'ids': ','.join(temp_list)},
+                                                            add_default_params=False)
+
+                    if 'errors' in response:
+                        if response['errors'][0]['code'] == 4010010:
+                            raise ValueError(f'Failed to authenticate to SentinelOne: {response}')
+                        else:
+                            raise ValueError(f'Error when authenticating to SentinelOne: {response}')
+
+                    for account in response:
+                        if account['id'] not in self._account_ids:
+                            self._account_ids.append(account['id'])
+
+                    counter = 0
+                    temp_list = []
+                i += 1
+
+            diff = list(set(account_ids) - set(self._account_ids))
+            if len(diff) > 0:
+                self.log.warning(f'Account IDs {",".join(diff)} not found.')
+
+        if account_names: # verify provided account names are valid
+            temp_account_name = list()
+            for name in account_names:
+                response = self._get_all_paginated_data(self._build_url('/web/api/v2.1/accounts'),
+                                                        params={'name': name},
```

Comment:
```suggestion
                                                        params={'states': "active", 'name': name},
```
only return active sites. any expired/deleted sites will show in the mismatch check and be logged.